doc/src/gotchas.rst

```diff
@@ -270,6 +270,33 @@ trick for getting tab completion in the regular Python console.
     See also `What is the best way to create symbols?
     <https://github.com/sympy/sympy/wiki/Faq>`_ in the FAQ.
 
+.. _symbols:
+
+Functions
+---------
+
+A function like ``f(x)`` can be created by defining the Function
+and the variable:
+
+  >>> from sympy import Function
+  >>> f = Function('f')
+  >>> x = Symbol('x')
+  >>> f(x)
+  f(x)
+
+If you assign ``f(x)`` to a Python variable `f` you will lose your
+ability to copy and paste that function or to create a function
+with a different argument: ``Function('f')`` is callable, but
+``Function('f')(x)`` is not:
```

Comment:
This doesn't explain what the real problem is. The whole point of this "gotcha" is that the error message isn't very helpful. So this should explain beyond just repeating it.

Reply:
Okay this was suggested here(https://github.com/sympy/sympy/issues/16836#issue-443970391) I will try to improve it.

Reply:
The idea is that if you make `f = Function('f')(x)` you can no longer write `f(x)` in an expression (you would have to write `f`) and you could not copy and past any expression showing `f(x)` as in `Eq(f(x), x + 2)`; you would have to re-enter that expression as `Eq(f, x + 2)`.

Reply:
So the current docstring is right? If yes I will solve the error and commit.